### PR TITLE
Mpi gpu test run

### PR DESCRIPTION
**Pull request by setmar**

### CODE REVIEW

File: gpu_ocean/SWESimulators/Simulator.py [hunk 1]
```diff
@@ -62,6 +62,8 @@ def __init__(self, \
                  write_netcdf, \
                  ignore_ghostcells, \
                  offset_x, offset_y, \
+                 ensemble_size, \
```
Comment (setmar) [inline gpu_ocean/SWESimulators/Simulator.py]: This will break the code if not all schemes are updated.

File: gpu_ocean/SWESimulators/Common.py [hunk 2]
```diff
@@ -62,9 +65,123 @@ def __exit__(self, *args):
         self.secs = self.end - self.start
         self.msecs = self.secs * 1000 # millisecs
         self.logger.log(self.log_level, "%s: %f ms", self.tag, self.msecs)
+
+
+class PopenFileBuffer(object):
```
Comment (babrodtk) [inline gpu_ocean/SWESimulators/Common.py]: All of this is from "my" repository, right?

File: gpu_ocean/prototypes/scripts/mpi_run_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+def my_exp(i, j):
+        x = dx*i - x_center
+        y = dy*j - y_center
+        return np.exp(-10*(x*x/(size*size)+y*y/(size*size)))
+        
+"""
+Initializes the KP simulator
+"""
+def initKP():
```
Comment (babrodtk) [inline gpu_ocean/prototypes/scripts/mpi_run_benchmark.py]: I Think we can skip all except CDKLM for now

File: gpu_ocean/SWESimulators/SimWriter.py [hunk 4]
```diff
@@ -47,6 +48,10 @@ def __init__(self, sim, filename=None, num_layers=1, staggered_grid=False, \
                  ignore_ghostcells=False, \
                  offset_x=0, offset_y=0):
 
+        # Parallel netCDF4 write?
```
Comment (babrodtk) [inline gpu_ocean/SWESimulators/SimWriter.py]: I don't think we will ahve time to support parallel write - so perhaps remove all of the if (write_parallel)-stuff?

File: gpu_ocean/SWESimulators/Common.py (hunk 2 above)
Comment (setmar) [reply gpu_ocean/SWESimulators/Common.py]: With some modifications, yes.

File: gpu_ocean/prototypes/scripts/mpi_run_benchmark.py (hunk 3 above)
Comment (setmar) [reply gpu_ocean/prototypes/scripts/mpi_run_benchmark.py]: (y)

File: gpu_ocean/SWESimulators/SimWriter.py (hunk 4 above)
Comment (setmar) [reply gpu_ocean/SWESimulators/SimWriter.py]: I agree. I will remove it if/when it requires maintenance.

**Pull request merged**